unified-runtime/source/adapters/hip/device.cpp

```diff
@@ -247,94 +247,46 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
     return ReturnValue(128u);
   }
   case UR_DEVICE_INFO_IMAGE2D_MAX_HEIGHT: {
-    // Take the smaller of maximum surface and maximum texture height.
     int TexHeight = 0;
     UR_CHECK_ERROR(hipDeviceGetAttribute(
         &TexHeight, hipDeviceAttributeMaxTexture2DHeight, hDevice->get()));
     assert(TexHeight >= 0);
-    int SurfHeight = 0;
-    UR_CHECK_ERROR(hipDeviceGetAttribute(
-        &SurfHeight, hipDeviceAttributeMaxTexture2DHeight, hDevice->get()));
-    assert(SurfHeight >= 0);
-
-    int Min = std::min(TexHeight, SurfHeight);
-
-    return ReturnValue(static_cast<size_t>(Min));
+    return ReturnValue(static_cast<size_t>(TexHeight));
   }
   case UR_DEVICE_INFO_IMAGE2D_MAX_WIDTH: {
-    // Take the smaller of maximum surface and maximum texture width.
     int TexWidth = 0;
     UR_CHECK_ERROR(hipDeviceGetAttribute(
         &TexWidth, hipDeviceAttributeMaxTexture2DWidth, hDevice->get()));
     assert(TexWidth >= 0);
-    int SurfWidth = 0;
-    UR_CHECK_ERROR(hipDeviceGetAttribute(
-        &SurfWidth, hipDeviceAttributeMaxTexture2DWidth, hDevice->get()));
-    assert(SurfWidth >= 0);
-
-    int Min = std::min(TexWidth, SurfWidth);
```

Comment:
The CUDA adapter also does this, but of course with different attributes.

This code kinda looks like it was a bad copy-paste, are you sure deleting is the right fix? It seems like there might be some hip attributes that give us the same thing as CUDA: [hipDeviceAttributeMaxSurface2D](https://rocm.docs.amd.com/projects/HIP/en/docs-6.0.0/doxygen/html/group___global_defs.html#ggacc0acd7b9bda126c6bb3dfd6e2796d7ca45e5f8d0e1b6b8ba58cf4b0f00b793b0)

Reply:
I have tested this attribute, it doesn't work on the HIP machine I have access to.

There are some attributes that are marked "CUDA only" in the documentation, meaning that they only work with the CUDA Runtime. While `hipDeviceAttributeMaxSurface2D` is not marked as "CUDA only", I believe that may be an error in the documentation. It could also be an error in the HIP drivers. Either way, the attribute doesn't work when I've tested it on a HIP Runtime, so I have omitted it.